File: Lenguajes Orientados a Objetos/Parcial 2 Actividad 4/C++/Empleado.cpp
#include "Empleado.h"

Empleado::Empleado(string nombre, string apellido, string nss) 
    : primerNombre(nombre), apellidoPaterno(apellido), numeroSeguroSocial(nss) {}

Empleado::~Empleado() {}

void Empleado::establecerPrimerNombre(string nombre) { primerNombre = nombre; }
string Empleado::obtenerPrimerNombre() const { return primerNombre; }

void Empleado::establecerApellidoPaterno(string apellido) { apellidoPaterno = apellido; }
string Empleado::obtenerApellidoPaterno() const { return apellidoPaterno; }

void Empleado::establecerNumeroSeguroSocial(string nss) { numeroSeguroSocial = nss; }
string Empleado::obtenerNumeroSeguroSocial() const { return numeroSeguroSocial; }

void Empleado::imprimir() const {
    cout << primerNombre << " " << apellidoPaterno 
         << "\nnumero de seguro social: " << numeroSeguroSocial;
}

File: Lenguajes Orientados a Objetos/Parcial 2 Actividad 4/Java/Empleado.java
// Fig. 10.4: Empleado.java
// La superclase abstracta Empleado.

public abstract class Empleado 
{
   private String primerNombre;
   private String apellidoPaterno;
   private String numeroSeguroSocial;

   // constructor con tres argumentos
   public Empleado( String nombre, String apellido, String nss )
   {
      primerNombre = nombre;
      apellidoPaterno = apellido;
      numeroSeguroSocial = nss;
   } // fin del constructor de Empleado con tres argumentos

   // establece el primer nombre
   public void establecerPrimerNombre( String nombre )
   {
      primerNombre = nombre;
   } // fin del método establecerPrimerNombre

   // devuelve el primer nombre
   public String obtenerPrimerNombre()
   {
      return primerNombre;
   } // fin del método obtenerPrimerNombre

   // establece el apellido paterno
   public void establecerApellidoPaterno( String apellido )
   {
      apellidoPaterno = apellido;
   } // fin del método establecerApellidoPaterno

   // devuelve el apellido paterno
   public String obtenerApellidoPaterno()
   {
      return apellidoPaterno;
   } // fin del método obtenerApellidoPaterno

   // establece el número de seguro social
   public void establecerNumeroSeguroSocial( String nss )
   {
      numeroSeguroSocial = nss; // debe validar
   } // fin del método establecerNumeroSeguroSocial

   // devuelve el número de seguro social
   public String obtenerNumeroSeguroSocial()
   {
      return numeroSeguroSocial;
   } // fin del método obtenerNumeroSeguroSocial

   // devuelve representación String de un objeto Empleado
   public String toString()
   {
      return String.format( "%s %s\nnumero de seguro social: %s", 
         obtenerPrimerNombre(), obtenerApellidoPaterno(), obtenerNumeroSeguroSocial() );
   } // fin del método toString

   // método abstracto sobrescrito por las subclases
   public abstract double ingresos(); // aquí no hay implementación
} // fin de la clase abstracta Empleado


/**************************************************************************
 * (C) Copyright 1992-2007 por Deitel & Associates, Inc. y                *
 * Pearson Education, Inc. Todos los derechos reservados.                 *
 *                                                                        *
 * RENUNCIA: Los autores y el editor de este libro han realizado su mejor *
 * esfuerzo para preparar este libro. Esto incluye el desarrollo, la      *
 * investigación y prueba de las teorías y programas para determinar su   *
 * efectividad. Los autores y el editor no hacen ninguna garantía de      *
 * ningún tipo, expresa o implícita, en relación con estos programas o    *
 * con la documentación contenida en estos libros. Los autores y el       *
 * editor no serán responsables en ningún caso por los daños consecuentes *
 * en conexión con, o que surjan de, el suministro, desempeño o uso de    *
 * estos programas.                                                       *
 *************************************************************************/